hebsafeharbor/identifier/signals/hebrew_city_recognizer.py

```diff
@@ -57,27 +59,138 @@ def analyze(
         if len(terms_offsets) == 0:
             return results
 
-        # iterate over the spaCy tokens, and find all the position indices for endorsing entities
-        geo_entities_position_set = set()
-        for entity in nlp_artifacts.entities:
-            if entity.label_ in self.endorsing_entities:
-                for char in range(entity.start_char, entity.end_char):
-                    geo_entities_position_set.add(char)
-
         # iterate over potential city offsets and endorse the results using overlap with geo-entities or support words
         for term_start, term_len in terms_offsets:
-            overlap_ratio = len(
-                set(range(term_start, term_start + term_len)).intersection(geo_entities_position_set)) / term_len
-            adjusted_score = self.DEFAULT_CONFIDENCE_LEVEL + (overlap_ratio * self.LOCATION_OVERLAP_FACTOR)
             result = RecognizerResult(
                 entity_type="CITY",
                 start=term_start,
                 end=term_start + term_len,
-                score=adjusted_score,
-                analysis_explanation=AnalysisExplanation(self.name, adjusted_score),
+                score=self.DEFAULT_CONFIDENCE_LEVEL,
+                analysis_explanation=AnalysisExplanation(self.name, self.DEFAULT_CONFIDENCE_LEVEL),
                 recognition_metadata={RecognizerResult.RECOGNIZER_NAME_KEY: self.name},
             )
 
             results.append(result)
 
+        results = self.enhance_using_other_entities(nlp_artifacts, results)
+
+        return results
+
+    def enhance_using_context(
+        self,
+        text: str,
+        raw_recognizer_results: List[RecognizerResult],
+        other_raw_recognizer_results: List[RecognizerResult],
+        nlp_artifacts: NlpArtifacts,
+        context: Optional[List[str]] = None,
+    ) -> List[RecognizerResult]:
+        """Overrides generic context enhancer with recognizer-specific.
+        Enhance confidence score using context of the entity.
+
+        in case a result score is boosted, derived class need to update
+        result.recognition_metadata[RecognizerResult.IS_SCORE_ENHANCED_BY_CONTEXT_KEY]
+
+        :param text: The actual text that was analyzed
+        :param raw_recognizer_results: This recognizer's results, to be updated
+        based on recognizer specific context.
+        :param other_raw_recognizer_results: Other recognizer results matched in
+        the given text to allow related entity context enhancement
+        :param nlp_artifacts: The nlp artifacts contains elements
+                              such as lemmatized tokens for better
+                              accuracy of the context enhancement process
+        :param context: list of context words
+        """
+
+        for result in raw_recognizer_results:
+            result.recognition_metadata[RecognizerResult.IS_SCORE_ENHANCED_BY_CONTEXT_KEY] = True
```

Comment:
Should this be changed every time? or only if the score is actually boosted? (line 117)

Reply:
This is to highlight that results were processed by custom enhancer, in our case we don't want to use the general enhancer, because it's less strict. I'l add a comment

Reply:
I mean, maybe we changed this to true, but then didn't really update the score due to the if statements down the pipe.